Feature/main pages clone

Mi4-04 opened the pull request.

Mi4-04 committed:
add main page

- change header
- add view block and list favorite movies

Mi4-04 committed:
add add page

Isidora25 commented on src/App.tsx:
```diff
@@ -1,24 +1,78 @@
     localStorage.getItem("isLoggedIn") === "1"
   );
 
+  const [genres, setGenres] = React.useState<IGenre[]>([]);
```
А ты же импортируешь напрямую useState, почему обращаешься через React?

Isidora25 commented on src/App.tsx:
```diff
@@ -1,24 +1,78 @@
+
+  const [view, setView] = React.useState<boolean>(false);
+
+  React.useEffect(() => {
```
Аналогично можно импортировать напрямую

Mi4-04 replied:
Так как мне лучше через React. или напрямую ?

Isidora25 commented on src/App.tsx:
```diff
@@ -1,24 +1,78 @@
 
+  const [genres, setGenres] = React.useState<IGenre[]>([]);
+
+  const [view, setView] = React.useState<boolean>(false);
```
А что значит эта переменная? вид = true

Mi4-04 replied:
Это флаг для того чтобы делать блочный или строчный вид списка

Isidora25 replied:
Ну вот не всегда же к тебе за справкой обращаться, согласись. Нужно назвать переменную так, чтобы было понятно, что она делает

Mi4-04 replied:
Понял, переименную

Isidora25 commented on src/utils/movies.ts:
```diff
@@ -34,5 +34,23 @@ export const getGenreList = async (): Promise<IGenre[]> => {
 }
+
+export const isValidFilmsIdAndFilms = (): void => {
+    if (localStorage.getItem("filmsIds") === null)
```
а что если там undefined?

Isidora25 commented on src/Pages/AddPage/component/Filtration/component/Popularity/index.tsx:
```diff
@@ -0,0 +1,32 @@
+        min={0}
+        max={10}
+        step={0.1}
+        value={props.voteAverage}
```
Лучше сразу сделать деструктуризацию пропсов в объявлении компонента [1]

Isidora25 commented on src/Pages/AddPage/component/Filtration/component/Popularity/index.tsx:
```diff
@@ -0,0 +1,32 @@
+}) => {
+  const { t } = useTranslation();
+
+  const handleChange = (e: any) => {
```
Не стоит использовать any без крайней необходимости. Здесь у тебя тип аргумента вполне известен [2]

Isidora25 commented on src/Pages/AddPage/component/Filtration/component/Popularity/index.tsx:
```diff
@@ -0,0 +1,32 @@
+      <Slider
+        min={0}
+        max={10}
+        step={0.1}
```
magic numbers. Старайся все числовые и строковые константы выносить в неизменяемые переменные с говорящим именем

Isidora25 commented on src/Pages/AddPage/component/Filtration/component/YearSelect/index.tsx:
```diff
@@ -0,0 +1,51 @@
+import Select from "@mui/material/Select";
+import { useTranslation } from "react-i18next";
+
+export const YearSelect = (props: {
```
см [1]

Isidora25 commented on src/Pages/AddPage/component/Filtration/component/YearSelect/index.tsx:
```diff
@@ -0,0 +1,51 @@
+
+export const YearSelect = (props: {
+  year: string;
+  setYear: (value: any) => void;
```
см [2]

Isidora25 commented on src/Pages/AddPage/component/Filtration/component/YearSelect/index.tsx:
```diff
@@ -0,0 +1,51 @@
+  setYear: (value: any) => void;
+}) => {
+  const { t } = useTranslation();
+  const [date, setDate] = React.useState<number[]>([]);
```
Как часто меняется список лет для отображения? Действительно здесь нужен стейт для хранения этой информации?

Isidora25 commented on src/Pages/AddPage/component/Filtration/component/YearSelect/index.tsx:
```diff
@@ -0,0 +1,51 @@
+    setDate([...date]);
+  }, []);
+
+  const handleChange = (e: any) => {
```
см [2]

Isidora25 commented on src/Pages/AddPage/component/Filtration/component/YearSelect/index.tsx:
```diff
@@ -0,0 +1,51 @@
+          onChange={handleChange}
+        >
+          {date?.map((dateYear) => {
+            return (
```
return не нужен

Isidora25 commented on src/Pages/AddPage/component/Filtration/index.tsx:
```diff
@@ -0,0 +1,46 @@
+import { YearSelect } from "@app/Pages/AddPage/component/Filtration/component/YearSelect";
+import { GenreList } from "@app/components/GenreList";
+
+export const Filtration = (props: {
```
см [1]

Isidora25 commented on src/Pages/AddPage/component/MoviesList/component/MoviesItemBlock/index.tsx:
```diff
@@ -0,0 +1,55 @@
+import { MovieLayout } from "./style";
+
+export const MoviesItemBlock = (props: {
+  film: any;
```
[2]

Isidora25 commented on src/Pages/AddPage/component/MoviesList/component/MoviesItemBlock/index.tsx:
```diff
@@ -0,0 +1,55 @@
+import { useTranslation } from "react-i18next";
+import { MovieLayout } from "./style";
+
+export const MoviesItemBlock = (props: {
```
[1]

Isidora25 commented on src/Pages/AddPage/component/MoviesList/component/MoviesItemBlock/index.tsx:
```diff
@@ -0,0 +1,55 @@
+      <Card sx={{ mt: 4, maxWidth: 250 }}>
+        <CardMedia
+          component="img"
+          image={`${URL_POST}${props.film.poster_path}`}
```
деструктурируй свойства объекта film

Isidora25 commented on src/Pages/AddPage/component/MoviesList/component/MoviesItemBlock/index.tsx:
```diff
@@ -0,0 +1,55 @@
+            color={
+              props.filmsIds.includes(props.film.id) ? "success" : "primary"
+            }
+            onClick={() => props.saveFilm(props.film.id)}
```
Лучше вынести в отдельную переменную функцию, а будет создаваться новая на каждый рендер [3]

Isidora25 commented on src/Pages/AddPage/component/MoviesList/component/MoviesItemList/index.tsx:
```diff
@@ -0,0 +1,48 @@
+      <ButtonPadLayout>
+        <Button
+          color={props.filmsIds.includes(props.film.id) ? "success" : "primary"}
+          onClick={() => props.saveFilm(props.film.id)}
```
[3]

Isidora25 commented on src/Pages/AddPage/component/MoviesList/index.tsx:
```diff
@@ -0,0 +1,73 @@
+
+  React.useEffect(() => {
+    setGenresId(
+      props.genres
```
Кусок кода очень плохо читается. Создай отдельную переменную, в которую ты будешь складывать айдишники жанров, а потом спокойно передавай ее в setGenresId

Isidora25 commented on src/Pages/AddPage/component/MoviesList/index.tsx:
```diff
@@ -0,0 +1,73 @@
+  React.useEffect(() => {
+    setGenresId(
+      props.genres
+        .filter((w, index) => {
```
А что ты пытаешься здесь этим фильтром найти так сложно? 
Для чего тебе w? 
О чем говорит переменная w? 

Если тебе нужно найти все жанры, которых isClick === true достаточно просто обратиться к этому полю

Isidora25 commented on src/Pages/AddPage/component/MoviesList/index.tsx:
```diff
@@ -0,0 +1,73 @@
+        .filter((w, index) => {
+          return props.genres[index].isClick;
+        })
+        .map((genre) => {
```
Если опять-таки я все правильно понимаю, то возвращать только айдишники жанров можно прямо из filter

Isidora25 commented on src/Pages/Main/component/FavoriteMovies/component/FavoriteMoviesItemLIst/style.ts:
```diff
@@ -0,0 +1,39 @@
+import styled from 'styled-components'
+import tw from 'twin.macro'
+
+interface IMovie {
```
Повторение кода. Вынеси этот интерфейс в специальный файл с типами и импортируй оттуда везде, где он нужен

Isidora25 commented on src/Pages/Main/component/FavoriteMovies/index.tsx:
```diff
@@ -0,0 +1,60 @@
+import { FavoriteMoviesItemList } from "./component/FavoriteMoviesItemLIst";
+
+export const FavoriteMovies = (props: { view: boolean }) => {
+  let [movies, setMovies] = React.useState<any[]>([]);
```
any

Isidora25 commented on src/Pages/Main/index.tsx:
```diff
@@ -0,0 +1,36 @@
+      <ButtonPad setView={props.setView} />
+
+      <ButtonLayout>
+        <Link to="/add">
```
Так button или link?

Mi4-04 replied:
ButtonLayout это из стилей а Link оборачивает Button

Isidora25 replied:
Ну тут нужно что-то одно, либо ссылка, которая выглядит как кнопка, либо кнопка, которая выглядит как ссылка. 
По смыслу, если элемент делает редирект куда-то - это ссылка. Если есть обработчик клика - это кнопка

Isidora25 commented on src/components/GenreList/index.tsx:
```diff
@@ -0,0 +1,47 @@
+
+export const GenreList = (props: {
+  genres: IGenre[];
+  setGenres: (value: any) => void;
```
any

Isidora25 commented on src/components/GenreList/index.tsx:
```diff
@@ -0,0 +1,47 @@
+}) => {
+  const [genresId, setGenresId] = React.useState<number[]>([]);
+
+  console.log("123", genresId);
```
Логи

Isidora25 commented on src/components/GenreList/index.tsx:
```diff
@@ -0,0 +1,47 @@
+  console.log("123", genresId);
+
+  React.useEffect(() => {
+    setGenresId(
```
где-то  я уже видела этот код
Если код повторяется, его следует выносить в отдельный компонент и переиспользовать

Isidora25 commented on src/components/GenreList/index.tsx:
```diff
@@ -0,0 +1,47 @@
+    props.setGenres([...props.genres]);
+  };
+
+  localStorage.setItem("genres", JSON.stringify(props.genres));
```
А для чего это сохраняется каждый рендер в LS?

Isidora25 commented on package.json:
```diff
@@ -8,6 +8,7 @@
     "@emotion/styled": "^11.6.0",
     "@material-ui/core": "^4.12.3",
     "@material-ui/icons": "^4.11.2",
+    "@mui/icons-material": "^5.4.4",
```
А почему часть кода написана с помощью material-ui, а другая с помощью tw? А где-то голый styled-components?

Mi4-04 replied:
Что именно поправить тут ?

Isidora25 replied:
Ну по ТЗ как стили должны быть написаны?

Mi4-04 committed:
fixed comments

-naming view on blockAndListView
-made distrukarization props
-fix different code

Mi4-04 committed:
fixed different components on distruktarization

Isidora25 commented on src/App.tsx:
```diff
@@ -16,14 +16,31 @@ const App = () => {
   const [genres, setGenres] = React.useState<IGenre[]>([]);
 
-  const [view, setView] = React.useState<boolean>(false);
+  const [blockAndListview, setBlockAndListview] =
```
А если blockAndListview = true - это какой вид? Блочный или список?

Mi4-04 replied:
Блочный

Isidora25 commented on src/App.tsx:
```diff
@@ -37,6 +54,8 @@ const App = () => {
     });
   }, []);
 
+  localStorage.setItem("genres", JSON.stringify(genres));
```
Сделай так, чтобы перезаписывалось только когда нужно

Mi4-04 committed:
fix coments
-delete button in link
-optimisation genres
-rename blockAndListView on blockView

Mi4-04 committed:
fix comments

Isidora25 commented on src/App.tsx:
```diff
@@ -54,6 +39,19 @@ const App = () => {
+        return genre.isClick;
+      })
+      .map((genre) => {
+        return genre.id;
```
FYI return не нужен

Isidora25 commented on src/App.tsx:
```diff
@@ -54,6 +39,19 @@ const App = () => {
+  React.useEffect(() => {
+    const isClicked = genres
+      .filter((genre) => {
+        return genre.isClick;
```
FYI return не нужен

Isidora25 commented on src/Pages/AddPage/component/MoviesList/component/MoviesItemBlock/index.tsx:
```diff
@@ -7,20 +8,25 @@ import Button from "@mui/material/Button";
+  );
+
+  const saveFilm = (id: number): void => {
+    let newfilmsIds = [...filmsIds, id];
```
newfilmsIds где-то еще изменяется? Если нет, то зачем  `let`?

Isidora25 commented on src/Pages/AddPage/component/MoviesList/component/MoviesItemList/index.tsx:
```diff
@@ -11,20 +12,25 @@ import {
 import Button from "@mui/material/Button";
+import { IMovies } from "@app/utils/movies";
 
 interface IMoviesItemList {
```
https://github.com/Mi4-04/movies-front/pull/7/commits/6569f6fb68758aac269632cfdce3415ae9040eb5#diff-7b47e345642209293d9afb7d8128c0f62efc7a4fe71787991f87fe7d096f29e1R13 дубликат интерфейса

Isidora25 commented on src/Pages/AddPage/component/MoviesList/component/MoviesItemList/index.tsx:
```diff
@@ -11,20 +12,25 @@ import {
+    JSON.parse(localStorage["filmsIds"])
+  );
+
+  const saveFilm = (id: number): void => {
```
А зачем ты вынес это в дочерние компоненты? 
Теперь код задублировался

Isidora25 commented on src/Pages/AddPage/component/MoviesList/index.tsx:
```diff
@@ -5,54 +5,42 @@ import { MoviesItemBlock } from "./component/MoviesItemBlock";
+  blockView,
+  genresId,
+}: IMoviesList) => {
+  const [movies, setMovies] = React.useState<any[]>([]);
```
а сюда `IMovies` не подойдет разве?

Mi4-04 committed:
fixed

Mi4-04 committed:
fixed

Mi4-04 committed:
fixed return filter
